sdonboarding.client/src/services/productServices.js

```diff
@@ -0,0 +1,55 @@
+/**This file contains the Product apis to send request to servers and get the response.
+ */
+
+import axios from "axios";
+
+export const productEndpoint = '/api/product';
+
+const productApiServices = {
+    async fetchProducts({ pageNumber, pageSize }) {
+        try {
+            const url = `${productEndpoint}?pageNumber=${pageNumber}&pageSize=${pageSize}`;
+
+            const response = await axios.get(url);
+            let data = response.data;
+            return data;
+        } catch (error) {
+            console.error("Fetch product failed!", error);
+            throw error;
+        }
+    },
+    async createProduct(newProduct) {
+        try {
+            const response = await axios.post(productEndpoint, newProduct);
+            return response.data;
+        } catch (error) {
+            console.error("create product failed!! ", error);
+            throw error;
+        }
+    },
+    async editProduct(editProduct) {
+        try {
+            const url = `${productEndpoint}/${editProduct.id}`;
```

Comment:
Just to make sure that you are processing and communication with your API correctly. Can you verify first if **id** contains an actual value before doing anything? Thanks